string id
string action_type
string param #TODO this maybe needs to be improved and specific action messages deal with the particular params for an action type
uint32 target_deadline
